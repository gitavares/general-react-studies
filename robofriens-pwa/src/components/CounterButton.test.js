import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

it('expect to render CounterButton component', () => {
    const mockColor = 'red'
    const component = shallow(<CounterButton color={mockColor} />);
    expect(component.debug()).toMatchSnapshot();
})

it('correctly increments the count', () => {
    const mockColor = 'red'
    const wrapper = shallow(<CounterButton color={mockColor} />);

    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.state()).toEqual({ count: 2 })
    wrapper.find('[id="counter"]').simulate('click')
    expect(wrapper.state()).toEqual({ count: 3 })
    wrapper.find('[id="counter"]').simulate('keypress')
    expect(wrapper.state()).toEqual({ count: 3 })
    expect(wrapper.props().color).toEqual('red')
})