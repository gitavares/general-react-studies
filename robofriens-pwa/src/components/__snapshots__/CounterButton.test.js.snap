// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
"<button id=\\"counter\\" color=\\"red\\" onClick={[Function: onClick]}>
  Count: 
  1
</button>"
`;
